Fix GH-1351: Make version dependent on xml call

mewtaylor opened the pull request:
Fixes #1351 by including a call to `version.xml` to get the most up-to-date offline build version, defaulting to the current version, `456.0.3`. This also fixes #1352 by switching the route to `/download`, with a redirect from `/scratch2download/`

### Test Cases ###
* Go to download page. All downloads for the offline editor should download
* Go to `/scratch2download` – you should be redirected to `/download`
* click on `offline editor` in the footer – you should be directed to `/download`

chrisgarrity commented on src/routes.json:
```diff
@@ -253,6 +253,12 @@
+        "name": "download-redirect",
+        "pattern": "^/scratch2download/?$",
+        "routeAlias": "/scratch2download",
+        "view": "/download"
```
This should be "redirect": "/download" not view.

mewtaylor replied:
thanks :)

mewtaylor commented:
1. @chrisgarrity I changed the redirect, should be good now
2. @cwillisf I changed the behavior to not show anything until it gets a response from `version.xml`, and also to display an error message if it can't get the `version.xml` file. Thoughts?
3. @thisandagain @carljbowman I put a background gray on the installation process (per #1350), but have not added in any icons – lemme know if I should :).

mewtaylor commented:
@TheGrits forgot to tag you on this – it'd be great to have you take a look if you have the time!

cwillisf commented:
Tested this and #1254 together on:
Windows 10.0.15063: Chrome 58.0.3029.110, Firefox 54.0, IE 11.413.15063.0, Edge 40.15063.0.0
macOS Sierra 10.12.5: Chrome 58.0.3029.110, Firefox ESR 38.7.1, Safari 10.1.1 (12603.2.4)

- [x] Download links work
- [x] Redirect `scratch2download` => `download` works
- [x] Retrieving version from XML works
- [x] No download links when version can't be read

I noticed #1381 and others noticed #1382, but those are not blocking issues.
